apps/sms/js/subject_composer.js

```diff
@@ -62,6 +62,13 @@
     }
   }
 
+  function updateDirection(value) {
+    /* jshint validthis: true */
+    var priv = privateMembers.get(this);
+
+    priv.input.dir = !value ? '': 'auto';
+  }
```

Comment:
I don't really like this :( do you think the badly positioned caret is really that bad?

At least add a comment to bug 1103011